### ISSUE TITLE
"Request access to the files using the Google Cloud Storage Browser" appears twice

### ISSUE THREAD

**Issue by tompollard** [MIT-LCP/physionet-build#1905]
There is a glitch in the "Files" section of the following project:
https://physionet.org/content/image-embeddings-mimic-cxr/1.0/#files

"Request access to the files using the Google Cloud Storage Browser." appears twice:

<img width="758" alt="Screenshot 2023-03-01 at 1 20 42 PM" src="https://user-images.githubusercontent.com/822601/222228237-3ec144d9-ba2d-4e80-bcda-cb383e95f909.png">

Possibly related to the restructuring of the project pages recently (@amitupreti) or perhaps related to the manual cloud upload for this project by @elfeto

**Comment by superryeti** [comment]
Thanks for reporting the issue @tompollard . I can work on this later today.

**Comment by superryeti** [comment]
should i assign myself to it? @tompollard

**Comment by superryeti** [comment]
Thank you

**Comment by elfeto** [comment]
I think its twice because one is for the ZIP and one is for the command to transfer the files from the bucket

<img width="1041" alt="Screenshot 2023-03-01 at 3 10 11 PM" src="https://user-images.githubusercontent.com/3857286/222242802-187ddb05-67a7-42c2-8a8b-ba6b2afffe5f.png">

**Comment by superryeti** [comment]
Here is my initial finding.

1. The text "Request access to the files using the Google Cloud Storage" appears only once in `published_project_data_access.html`

```
 {% if data_access %}
    {% for item in data_access %}
      {% if item.platform == 1 %} {# ID for AWS open data #}
        <li>Access using AWS <a href="{{item.location}}">Open Data repository</a></li>
      {% elif item.platform == 2 %} {# ID for AWS cloud bucket #}
        <li><a href="{% url 'published_project_request_access' project.slug project.version 2 %}">Request access</a> to the data using AWS command line tools: <pre class="shell-command">aws s3 sync {{item.location}} DESTINATION</pre></li>
      {% elif item.platform == 3 %} {# ID for Google cloud bucket email #}
        {% if project.gcp and project.gcp.sent_files %}
            <li><a href="{% url 'published_project_request_access' project.slug project.version 3 %}">Request access</a> to the files using the <a href="https://console.cloud.google.com/storage/browser/{{ project.gcp.bucket_name }}/">Google Cloud Storage Browser</a>. Login with a Google account is required.</li>
        {% endif %}

      {% elif item.platform == 4 %} {# ID for google BigQuery email #}
        <li><a href="{% url 'published_project_request_access' project.slug project.version 4 %}">Request access</a> using Google BigQuery.</li>
      {% endif %}
    {% endfor %}
{% endif %}

```

Here the text should only appear for platform type 3 which is `gcp-bucket`


2. The template `published_project_data_access.html` is only included once in `published_project.html`  as `{% include "project/published_project_data_access.html" %}`. Since this statement is not inside any iterator. I dont think the the line is appearing twice because of some sort of bug on template.

3. So the conclusion is that there are two entries for `gcp-bucket`   on `DataAccess` and hence the for loop is executing twoce and showing the access mechanism twice. @tompollard  Could you please check if this is the case? If not i can dig deeper.

**Comment by superryeti** [comment]
> I think its twice because one is for the ZIP and one is for the command to transfer the files from the bucket <img width="1041" alt="Screenshot 2023-03-01 at 3 10 11 PM" src="https://user-images.githubusercontent.com/3857286/222242802-187ddb05-67a7-42c2-8a8b-ba6b2afffe5f.png">

@elfeto quick question. I think this might not be the case(though i am probably wrong)

Because on Tom's screenshot; if you compare word by word, the duplicated lines are identical, but in your screenshot, they are different.

**Comment by bemoody** [comment]
I am really confused about what's going on in published_project.html.  But the reason there are two "request access" links for this particular project is that this particular project has two DataAccesses (pointing to different authorization group addresses for the GCP bucket.)

**Comment by briangow** [comment]
It looks someone cleaned up the `DataAccess` object so this now appears only once.

**Comment by tompollard** [comment]
I did come across a duplicate `DataAccess` object while implementing the data security program thing. I deleted it! I assume it was this project but can't say for sure without digging through my notes (which I won't do unless someone asks).
